code/modules/fish/fishtank.dm

```diff
@@ -0,0 +1,701 @@
+							M.visible_message("[M.name] caught and devoured a live fish!", "You catch and devour a live fish, yum!")
+							kill_fish()						//Kill a random fish
+							M.health = M.maxHealth			//Eating fish heals the predator
+				else
+					usr << "There are no fish in [src]!"
+		else
+			attack_generic(M, M.harm_intent_damage)
+	else if(istype(M, /mob/living/simple_animal/hostile/bear))
+		if(M.a_intent == "help")							//Bears can try to fish in open tanks on help intent
+			if(lid_switch)									//Can't fish in a closed tank. Fishbowls are ALWAYS open.
+				M.visible_message("[M.name] scrapes it's claws along \the [src]'s lid.", "The lid is closed, so you scrape your claws against \the [src]'s lid.")
+			else
+				if(fish_count)								//Tank must actually have fish to try catching one
+					M.visible_message("[M.name] reaches into  \the[src] and attempts to fish through the opening!", "You reach into \the [src] and begin fishing through the opening!")
+					spawn(5)
+						if(water_level && prob(5))			//Bears are good at catching fish, only a 5% chance to fail
+							M.visible_message("[M.name] swipes at the water!", "You just barely missed that fish!")
+						else								//No water or didn't slip, get that fish!
+							M.visible_message("[M.name] caught and devoured a live fish!", "You catch and devour a live fish, yum!")
+							kill_fish()						//Kill a random fish
+							M.health = M.maxHealth			//Eating fish heals the predator
+				else
+					usr << "There are no fish in [src]!"
+		else
+			attack_generic(M, M.harm_intent_damage)
+	else
+		if(M.melee_damage_upper > 0)						//If the simple_animal has a melee_damage_upper defined, use that for the damage
+			attack_generic(M, M.melee_damage_upper)
+		else if(M.a_intent == "harm")						//Let any simple_animal try to break tanks when on harm intent
+			if(M.harm_intent_damage <= 0) return			//If it doesn't do damage, don't bother with the attack
+			attack_generic(M, M.harm_intent_damage)
+	check_health()
+
+/obj/machinery/fishtank/attack_alien(mob/living/user as mob)
+	if(islarva(user)) return
+	attack_generic(user, 15)
+
+/obj/machinery/fishtank/attack_slime(mob/living/user as mob)
+	var/mob/living/carbon/slime/S = user
+	if (!S.is_adult)
+		return
+	attack_generic(user, rand(10, 15))
+
+/obj/machinery/fishtank/attack_hand(mob/user as mob)
+	if(HULK in user.mutations)
+		user.say(pick(";RAAAAAAAARGH!", ";HNNNNNNNNNGGGGGGH!", ";GWAAAAAAAARRRHHH!", "NNNNNNNNGGGGGGGGHH!", ";AAAAAAARRRGH!"))
+		user.visible_message("<span class='danger'>[user] smashes through [src]!</span>")
+		destroy()
+	else if (usr.a_intent == "harm")
+		user.changeNext_move(CLICK_CD_MELEE)
+		playsound(get_turf(src), 'sound/effects/glassknock.ogg', 80, 1)
+		usr.visible_message("\red [usr.name] bangs against the [src.name]!", \
+							"\red You bang against the [src.name]!", \
+							"You hear a banging sound.")
+	else
+		user.changeNext_move(CLICK_CD_MELEE)
```

Comment:
Not really necessary, I think. They are just tapping on it, not doing a hostile action.